app/assets/javascripts/discourse-markdown-it/src/features/image-controls.js

```diff
@@ -31,7 +31,11 @@ function isUpload(token) {
 }
 
 function hasMetadata(token) {
-  return token.content.match(/(\d{1,4}x\d{1,4})/);
+  return (
+    token.content
+      .split("|")
+      .find((part) => /^\d{1,4}x\d{1,4}(,\s*\d{1,3}%)?$/.test(part)) || null
+  );
```

Comment:
Since this function is supposed to return a bool can we do this instead?

```suggestion
  return !!token.content
    .split("|")
    .find((part) => /^\d{1,4}x\d{1,4}(,\s*\d{1,3}%)?$/.test(part));
```